src/lib/structures/expr.ml

```diff
@@ -2169,39 +2184,53 @@ module Triggers = struct
     let l = STRS.elements s in (* remove redundancies in old l *)
     SLLT.elements (parties_rec (SLLT.empty, SLLT.empty) l)
 
+  (* Simplify the multi-trigger [l] by removing a pattern [p] in [l] if
+     its set of free (term and type) variables:
+     - isn't maximal (for the inclusion) among the set of free variables of
+         patterns in [l];
+     - contains the union of [bv_a] and [vty_a];
+     - is disjoint with the union of [bv_a] and [vty_a]. *)
```

Comment:
I don't understand how these two conditions can hold at the same time, and I thought this was an AND-list. Is it an OR-list?

Reply:
It's a OR-list.